### PR TITLE
Feature/sc 26301/add tests vpc module

### PR DESCRIPTION
**Pull request by JosueMolinaMorales**
Adds tests for the vpc module

### CODE REVIEW

File: modules/vpc/main.tf [hunk 1]
```diff
@@ -81,7 +81,7 @@ data "aws_availability_zones" "current" {
 # ------------------------------------------------------------
 
 locals {
-  num_availability_zones = var.num_availability_zones == null ? length(data.aws_availability_zones.current) : var.num_availability_zones
+  num_availability_zones = var.num_availability_zones == null ? length(data.aws_availability_zones.current.names) : var.num_availability_zones
```
Comment (Cwagne17) [inline modules/vpc/main.tf:84]: If we have to reference `.names` here. Does this also need to be referenced later in the config where the actual availability zone name is being used?

File: test/modules/vpc.go [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+	awsRegion := aws.GetRandomStableRegion(t, []string{"us-east-1", "us-east-2"}, nil)
+	test_structure.SaveString(t, workingDir, "awsRegion", awsRegion)
+
+	// Construct the terraform options with default retryable errors to handle the most common retryable errors in
+	// terraform testing.
+	terraformOptions := terraform.WithDefaultRetryableErrors(t, &terraform.Options{
+		// The path to where our Terraform code is located
+		TerraformDir: workingDir,
+		Vars: map[string]interface{}{
+			"region": awsRegion,
```
Comment (Cwagne17) [inline test/modules/vpc.go:29]: Do we need to namespace the name of the VPC to avoid conflict?

File: test/modules/vpc.go [hunk 3]
```diff
@@ -0,0 +1,262 @@
...
+	// Assert that the VPC has no private subnets
+	subnets, err := ec2Client.DescribeSubnets(&ec2.DescribeSubnetsInput{
+		Filters: []*ec2.Filter{
+			{
+				Name:   aws_sdk.String("vpc-id"),
+				Values: []*string{aws_sdk.String(vpcID)},
+			},
+			{
+				Name:   aws_sdk.String("tag:Name"),
+				Values: []*string{aws_sdk.String("Private Subnet")},
```
Comment (Cwagne17) [inline test/modules/vpc.go:66]: The Name tag for a private subnet is formatted like `"${var.vpc_name}-private-%s"` likewise, the public subnet is formatted like `"${var.vpc_name}-public-%s"` where the `%s` is the availability zone name.

File: test/modules/vpc.go [hunk 4]
```diff
@@ -0,0 +1,262 @@
...
+	assertVpcHasCorrectNumberOfSubnets(t, terraformOptions, awsRegion, "public_subnet_ids", numAzs)
+
+	// Assert that the Public CIDR blocks are computed correctly
+	assertPublicCidrBlocksAreCorrect(t, terraformOptions, numAzs)
+
+	// Assert that the Public Route tables direct traffic to the Internet Gateway
+	assertPublicRouteTablesHaveCorrectRoutes(t, terraformOptions, ec2Client, vpcID)
+
+	// Assert that the Public NACL is configured Correctly
+	acls, err := ec2Client.DescribeNetworkAcls(&ec2.DescribeNetworkAclsInput{})
```
Comment (Cwagne17) [inline test/modules/vpc.go:103]: Currently our NACL are allowing allow ingress/egress.

Comment (Cwagne17) [reply test/modules/vpc.go:103]: So its ok if we don't check this, as long as they exist for now.

File: test/modules/vpc.go [hunk 5]
```diff
@@ -0,0 +1,262 @@
...
+			{
+				Name:   aws_sdk.String("attachment.vpc-id"),
+				Values: []*string{aws_sdk.String(vpcID)},
+			},
+		},
+	})
+	assert.NoError(t, err, "Error describing Internet Gateways")
+
+	// Assert There are Internet Gateways
+	assert.NotEqual(t, 0, len(igtw.InternetGateways), "Expected at least 1 Internet Gateway, got 0")
```
Comment (Cwagne17) [inline test/modules/vpc.go:166]: Can we add newlines between the assertions for readability?

File: test/modules/vpc.go [hunk 6]
```diff
@@ -0,0 +1,262 @@
...
+	publicCidrBlocks := terraform.OutputList(t, terraformOptions, "public_subnet_cidr_blocks")
+	assert.Equal(t, numAzs, len(publicCidrBlocks), "Expected %d Public CIDR Blocks, got %d", numAzs, (publicCidrBlocks))
+
+	// Should have format 10.0.*.0/24
+	for i, cidr := range publicCidrBlocks {
+		assert.Equal(t, fmt.Sprintf("10.0.%d.0/24", i+1), cidr)
+	}
+}
+
+func assertCidrBlocksAreCorrect(t *testing.T, terraformOptions *terraform.Options, numAzs int) {
```
Comment (Cwagne17) [inline test/modules/vpc.go:242]: Should this be renamed to `assertPrivateCidrBlocksAreCorrect`?

**Review by Cwagne17 — CHANGES_REQUESTED** [review]
Just a few questions & requests. The testing is looking really good. Could we also add doc comments to functions? https://tip.golang.org/doc/comment

**Comment by JosueMolinaMorales** [comment]
> Looks good, but tests are failing.

You're failing

**Pull request merged**